clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -91,8 +122,22 @@ public NetworkClient(Selectable selector,
                          int socketReceiveBuffer,
                          int requestTimeoutMs,
                          Time time) {
-        this(null, metadata, selector, clientId, maxInFlightRequestsPerConnection,
-                reconnectBackoffMs, socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time);
+        this(NO_METADATA_UPDATER, metadata, selector, clientId, maxInFlightRequestsPerConnection,
+                reconnectBackoffMs, socketSendBuffer, socketReceiveBuffer, requestTimeoutMs, time, NO_APIS);
+    }
+
+    public NetworkClient(Selectable selector,
+                         MetadataUpdater metadataUpdater,
+                         String clientId,
+                         int maxInFlightRequestsPerConnection,
+                         long reconnectBackoffMs,
+                         int socketSendBuffer,
+                         int socketReceiveBuffer,
+                         int requestTimeoutMs,
+                         Time time,
+                         Collection<ApiVersionsResponse.ApiVersion> apis) {
```

Comment:
Hmm.  "java.util.Collection" could be something unsorted like a List.  How about making this a map from ApiKey to ApiVersion so that we can compare it with the server response more efficiently?